Azure/communication-ui-library-android
# [Refactor] Refactor Confirm Leave Overlay View Logic

## Purpose

* Refactor confirm leave overlay logic into ViewModel

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

1. Join the Group Call/ Teams Meeting
2. Press the Hang Up Button
3. Check Confirm Leave Overlay Behaviour

## What to Check
Verify that the following are valid
* Confirm Leave Overlay Behaviour does not Change
